x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3265,6 +3292,26 @@ function getUnsnoozeAttributes(attributes: RawRule, scheduleIds?: string[]) {
   };
 }
 
+function getBulkUnsnoozeAttributes(attributes: RawRule, scheduleIds?: string[]) {
+  // Bulk removing snooze schedules, don't touch the current snooze/indefinite snooze
+  if (scheduleIds) {
+    const newSchedules = clearScheduledSnoozesById(attributes, scheduleIds);
+    const unscheduledSnooze =
+      attributes.snoozeSchedule?.filter((s) => typeof s.id === 'undefined') || [];
+
+    return {
+      snoozeSchedule: [...unscheduledSnooze, ...newSchedules],
```

Comment:
@JiaweiWu Not sure I understand what going on here. Can you help me?
`clearScheduledSnoozesById` just will filter out from `attributes.snoozeSchedule` all scheduled snoozes which ids matches scheduleIds.
And `unscheduledSnooze` are scheduled snoozes from the same `attributes.snoozeSchedule`, but which have `undefined` for id.
So, are `newSchedules` already includes `unscheduledSnooze` in that case. Or I'm missing something?

Reply:
I saw the same thing and I talked to @JiaweiWu and he add a comment, he did not want to change the name because it is already name like this in different function but we did add a comment.

<img width="689" alt="image" src="https://user-images.githubusercontent.com/189600/192607670-2cca66a8-a8ba-4bfd-a3ac-289663d96533.png">